avocado/utils/nfs.py

```diff
@@ -0,0 +1,349 @@
+        """
+        ssh_cmd = "ssh %s@%s " % (self.ssh_user, self.nfs_client_ip)
+        find_mountpoint_cmd = "mount | grep -E '.*%s.*%s.*'" % (self.mount_src,
+                                                                self.mount_dir)
+        cmd = ssh_cmd + "'%s'" % find_mountpoint_cmd
+        logging.debug("The command: %s", cmd)
+        status, output = commands.getstatusoutput(cmd)
+        if status:
+            logging.debug("The command result: <%s:%s>", status, output)
+            return False
+
+        return True
+
+    def setup(self):
+        """
+        Setup NFS client.
+        """
+        # Mount sharing directory to local host
+        # it has been covered by class Nfs
+
+        # Mount sharing directory to remote host
+        if self.remote_nfs_mount == "yes":
+            self.setup_remote()
+
+    def umount(self):
+        """
+        Unmount the mount directory in remote host.
+        """
+        ssh_cmd = "ssh %s@%s " % (self.ssh_user, self.nfs_client_ip)
+        logging.debug("Umount %s from %s" %
+                      (self.mount_dir, self.nfs_client_ip))
+        umount_cmd = ssh_cmd + "'umount -l %s'" % self.mount_dir
+        try:
+            process.system(umount_cmd, verbose=True)
+        except process.CmdError:
+            logging.warn("Failed to run: %s" % umount_cmd)
+
+    def cleanup(self, ssh_auto_recover=True):
+        """
+        Cleanup NFS client.
+        """
+        self.umount()
+        ssh_cmd = "ssh %s@%s " % (self.ssh_user, self.nfs_client_ip)
+        if self.mkdir_mount_remote:
+            rmdir_cmd = ssh_cmd + "'rm -rf %s'" % self.mount_dir
+            try:
+                process.system(rmdir_cmd, verbose=True)
+            except process.CmdError:
+                logging.warn("Failed to run: %s" % rmdir_cmd)
+
+        if self.is_mounted():
+            logging.warn("Failed to umount %s" % self.mount_dir)
+
+        # Recover SSH connection
+        self.ssh_obj.auto_recover = ssh_auto_recover
+        del self.ssh_obj
+
+    def setup_remote(self):
+        """
+        Mount sharing directory to remote host.
+        """
+        ssh_cmd = "ssh %s@%s " % (self.ssh_user, self.nfs_client_ip)
+        check_mount_dir_cmd = ssh_cmd + "'ls -d %s'" % self.mount_dir
+        logging.debug("To check if the %s exists", self.mount_dir)
+        output = commands.getoutput(check_mount_dir_cmd)
+        if re.findall("No such file or directory", output, re.M):
+            mkdir_cmd = ssh_cmd + "'mkdir -p %s'" % self.mount_dir
+            logging.debug("Prepare to create %s", self.mount_dir)
+            status, output = commands.getstatusoutput(mkdir_cmd)
+            if status != 0:
+                logging.warn("Failed to run %s: %s" % (mkdir_cmd, output))
+            self.mkdir_mount_remote = True
+
+        self.mount_src = "%s:%s" % (self.nfs_server_ip, self.mount_src)
+        logging.debug("Mount %s to %s" % (self.mount_src, self.mount_dir))
```

Comment:
The `process.system` calls are usually logged, so some of those messages could be removed.